src/js/shipping-workshop-block/block.js

```diff
@@ -77,6 +77,12 @@ export const Block = ( { checkoutExtensionData, extensions } ) => {
 		 * [frontend-step-02-extra-credit-1]
 		 * 💰 Extra credit: Ensure the `setExtensionData` function is not called multiple times. We
 		 * can use the `debouncedSetExtensionData` function for this. The API is the same.
+		 * 
+		 * Debounced function doesn't seem necessary for the Select fields – a user could hardly manage
+		 * to change options faster than once per second. At some moment started throwing an error for
+		 * 'extensions' parameter (probably, extension data was not passed properly).
+		 * Didn't find the reason yet.
+		 * 
```

Comment:
I don't think this is necessary to include in the comment. We could leave it as-is and look into fixing the suggestion of adding a debounced function instead.

Reply:
Do you have any idea why a debounced function could suddenly become problematic? Most of the time, it works perfectly fine. However, at one point, it started behaving erratically, even though I haven’t been able to reproduce the issue since.

Reply:
Where I've suggested to remove the `if` - it's the spoiler .md, while in the code in this repo contains all steps already implemented (unlike the 'starter' one, where one should go through all steps to make it functional). This was one of discoveries I wanted to report (like the whole code is complete, but here the task is not implemented). And I've stuck to this because of that moment of erratical behaviour. Maybe I was so tired that I believed I've checked dependencies, while in reality I didn't, so far I can't reproduce it. 

My opinion - make it debounced, to have it aligned with spoilers and everything. 

I'm going to use this workshop as a basement for my own extension, so if I will run into this problem again – I'll do my best to find out how to reproduce it it will drop you a message.